File: codegen/results/v002/qwen3.5_4b-nvfp4_think/20260412_083548/typescript/003_fibonacci/solution.ts
import { argv } from 'process';

// Parse command-line argument - default to 0 if not provided
const N = parseInt(process.argv[2] || '0');

// Check if N < 1, exit without output
if (N < 1) {
  process.exit(0);
}

// Fibonacci sequence: starts with 1, 1, then each next is sum of previous two
let prev = 1;     // First number in sequence
let curr = 1;     // Second number in sequence

// Print first Fibonacci number (always 1 since N >= 1)
console.log(prev);

// Continue generating and printing numbers as long as curr <= N
while (curr <= N) {
  console.log(curr);
  
  // Calculate next Fibonacci number
  const next = prev + curr;
  
  // Update for next iteration
  prev = curr;
  curr = next;
}

// This stops when the next Fibonacci number would exceed N

File: codegen/results/v002/qwen3.5_4b-nvfp4_think/20260412_083548/python/003_fibonacci/solution.py
import sys

def main():
    # Get the integer argument from command line
    if len(sys.argv) != 2:
        print("Usage: python script.py <integer>")
        return

    try:
        n = int(sys.argv[1])
    except ValueError:
        print("Usage: python script.py <integer>")
        return

    # If N < 1, print nothing
    if n < 1:
        return

    # Generate Fibonacci numbers up to N
    a, b = 1, 1
    fib_count = [1]

    while b <= n:
        a, b = b, a + b
        if a > n:
            break
        fib_count.append(a)
        fib_count.append(b) if b <= n else None

    # Print each Fibonacci number on its own line
    for num in fib_count:
        print(num)

if __name__ == "__main__":
    main()